src/main/java/com/iota/iri/service/snapshot/conditions/SnapshotSizeCondition.java

```diff
@@ -78,11 +92,29 @@ public boolean shouldTakeSnapshot(boolean isInSync) {
         
         // Update the last size
         lastSize = size;
+        
         return size > maxSize;
     }
 
     @Override
     public int getSnapshotStartingMilestone() throws SnapshotException {
-        return snapshotProvider.getInitialSnapshot().getIndex() + MILESTONES;
+        // Snapshot by size doesn't need a recent snapshot, we calculate here so we can reuse it when we call getSnapshotPruningMilestone.
+        // This will not go below the allowed snapshot depth due to the other conditions
+        return -1;
+    }
+
+    @Override
+    public int getSnapshotPruningMilestone() throws SnapshotException {
+        int initialIndex = -1;
+        try {
+            Pair<Indexable, Persistable> ms = tangle.getFirst(Milestone.class, IntegerIndex.class);
+            initialIndex = ((IntegerIndex)ms.low).getValue();
+        } catch (Exception e) {
+            log.error("failed to find oldest milestone", e);
+            // This should never fail, if it does, we just don't prune
+            return -1;
+        }
+        
+        return initialIndex + MILESTONES;
```

Comment:
I don't understand, you just always prune 5 milestones above?
This is how you get a minimal DB?
Shouldn't it be whatever max depth is defined to be at least?

Reply:
max depth is checked on LSManagerImpl

Reply:
I explained the case of magic 5 in the notes on the top, i dont like it either..

Its a SnapshotCondition because it handles a condition in which we take a snapshot, which sometimes links to pruning as well, but snapshot is the main thing. We could split it off again but that would just be a lot of almost empty classes causing confusion since its all too spread out.

I do agree that the shouldTakeSnapshot check seems weird to do in pruningHandle, i could make the same one for shouldDoPruning, and just the same code called in the class for clarity?

Reply:
I think I am going to do it because looking again at your code I think it is buggy

If we take an LS when `shouldDoSnapshot` returns `true` then it will never return `true` again for pruning...

Reply:
ok